### PR TITLE
cmake: Fix for macOS Lion

### PR DESCRIPTION
**Pull request by lemzwerg**

### CODE REVIEW

**Comment by lemzwerg** [comment]
If this is so, you should definitely change the text that gets output in case of error!  It says that people should provide pull requests.  I did exactly that.

On the other hand: My changes are *very* small and don't hurt at all.  Additionally, TigerBrew doesn't actively support Lion, AFAICS.  In other words, people who want to stay on Lion (for whatever reason) had a working Homebrew, say, two years ago, and are now completely out of luck.

**Comment by SMillerDev** [comment]
> In other words, people who want to stay on Lion (for whatever reason) had a working Homebrew, say, two years ago, and are now completely out of luck.

Yes, but that's the case with most software. If you don't update, support for your version will be dropped. And lion, in your suggestion, would be only supported by homebrew.

**Comment by lemzwerg** [comment]
BTW, Macports still supports Lion, and, AFAICS, the only reason Homebrew doesn't support older versions is the lacking ability to test.  Please correct me if I'm wrong – there might be ideological reasons that I'm not aware of.

Actively supporting only the last three OS releases is a reasonable assumption, and I agree that the Homebrew people have to define targets where to put the efforts.  Speaking as the FreeType maintainer, however, I know that under some circumstances OS (or compiler) updates are not always possible...

So please do a decision whether you *want* pull requests for older OS X releases.  Otherwise I urge you to adjust the warning text so that people like me avoid hours of time for wasted development.

File: Formula/cmake.rb [hunk 1]
```diff
@@ -27,6 +27,13 @@ class Cmake < Formula
   def install
     ENV.cxx11 if MacOS.version < :mavericks
 
+    # Avoid the following compiler error:
+    # SecKeychain.h:102:46: error: shift expression '(1853123693 << 8)' overflows
+    if MacOS.version <= :lion
+      ENV.append "CFLAGS", "-fpermissive"
+      ENV.append "CXXFLAGS", "-fpermissive"
+    end
```
Comment (sjackman) [inline Formula/cmake.rb]: I suggest (for style)
```ruby
ENV.append_to_cflags "-fpermissive" if MacOS.version <= :lion
```

**Review by sjackman — DISMISSED** [review]

File: Formula/cmake.rb (hunk 1 above)
Comment (lemzwerg) [reply Formula/cmake.rb]: Your suggested single line works for both `CFLAGS` and `CXXFLAGS`?

Comment (sjackman) [reply Formula/cmake.rb]: Yes.
https://github.com/Homebrew/brew/blob/8778cc308ac6256c2e257962e1783d7f425ba6f1/Library/Homebrew/extend/ENV/shared.rb#L14
https://github.com/Homebrew/brew/blob/8778cc308ac6256c2e257962e1783d7f425ba6f1/Library/Homebrew/extend/ENV/shared.rb#L49-L51

Comment (lemzwerg) [reply Formula/cmake.rb]: Thanks.  Change pushed.

**Pull request merged**